### PR TITLE
fix: Fix handling of imports in protocol-buffer definitions

### LINKED ISSUE #10794: Telegraf not able to read multiple .proto files
### Relevant telegraf.conf

```toml
[agent]
## Default data collection interval for all inputs
  interval = "1s"
  ## Log at debug level.
  debug = true
  ## Log only error level messages.
  quiet = false


[[inputs.mqtt_consumer]]
  servers = ["mqtt+ssl://awshostname:8883"]
  qos = 0
  ## Topics that will be subscribed to.
  topics = [
    "sci-topic.sc.proxy1",
    "sci-topic/sc/proxy1"
  ]
  connection_timeout = "30s"

  ## If unset, a random client ID will be generated.
  client_id = "telegraf1"

  ## Username and password to connect MQTT server.
  username = "uername"
  password = "password"

  data_format = "xpath_protobuf"
  xpath_protobuf_type = "com.ruckus.cloud.lte.sciproxy.gpb.SciMessage"
  xpath_protobuf_file = "/etc/telegraf/cbrs/protofiles/sci_message.proto"
  xpath_print_document = true
```

### Logs from Telegraf

```
[telegraf] Error running agent: Error loading config file /etc/telegraf/cbrs/telegraf.conf: error parsing mqtt_consumer, parsing protocol-buffer definition in "/etc/telegraf/cbrs/protofiles/sci_message.proto" failed: /etc/telegraf/cbrs/protofiles/sci_message.proto:14:8: open ap_report.proto: no such file or directory
2022-03-08T06:20:16Z I! Starting Telegraf 1.20.0
2022-03-08T06:20:16Z E! [telegraf] Error running agent: Error loading config file /etc/telegraf/cbrs/telegraf.conf: error parsing mqtt_consumer, parsing protocol-buffer definition in "/etc/telegraf/cbrs/protofiles/sci_message.proto" failed: /etc/telegraf/cbrs/protofiles/sci_message.proto:14:8: open ap_report.proto: no such file or directory
2022-03-08T06:20:17Z I! Starting Telegraf 1.20.0
```

### System info

telegraf 1.20.0 docker image

### Docker

_No response_

### Steps to reproduce

1. create at least 2 proto files
2. Try to import the second file into the first file and use the second file message as a field in the first file. 
3. Now pass the first file in telegraf.conf
...
I am providing the steps to create the files as well

### first.proto
```protobuf
import "second.proto";

message first {  
  string query = 1;
  int32 page_number = 2;
  int32 result_per_page = 3;
  
 second Second2 = 10;
  
}
```

### second.proto
```protobuf
message second {
  string url = 1;
  string title = 2;
  repeated string snippets = 3;
}

message second2{

  uint32 var1 = 1;
  uint64 var2 = 2;

    repeated second Second 	= 10;
}
```

### Expected behavior

First one proto file should be able to import the second proto file and able to decode the corresponding message.

### Actual behavior

it's giving the import error saying `no such file or directory`

### Additional info

removing the import and corresponding message proto file works fine.
The error only comes when trying to import other proto files.

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #10794

Currently, the XPath parser fails when confronted with protocol-buffer definitions that import other definitions and are not in the current working directory. To solve the issue, we add a new option (`xpath_protobuf_import_paths`) to allow the user to specify import paths that are used to look up specified and imported definitions. If, for example, you do have files `A.proto` and `B.proto` within a package `foo` that are located in `/data/my_proto/files` you can use
```toml
  data_format = "xpath_protobuf"
  xpath_protobuf_file = "A.proto"
  xpath_protobuf_type = "foo.Measurement"
  xpath_protobuf_import_paths = [".", "/data/my_proto_files"]
 ...
 ```
to specify the required information to the parser.

### CODE REVIEW

File: plugins/parsers/xpath/README.md [hunk 1]
```diff
@@ -13,11 +13,51 @@ For supported XPath functions check [the underlying XPath library][xpath lib].
...
+
+#### `xpath_protobuf_type` (mandatory)
+
+This option contains the top-level message file to use for deserializing the data to be parsed. Usually, this is constructed from the `package` name in the protocol-buffer definition file and the `message` name as `<package name>.<message name>`.
+
+#### `xpath_protobuf_import_paths` (optional)
+
+In case you import other protocol-buffer definitions within your `.proto` file (i.e. you use the `import` statement) you can use this option to specify paths to search for the imported definition file(s). By default the imports are only searched in `.` which is the current-working-directory, i.e. usually the directory you are in when starting telegraf.
+
+Imagine you do have multiple protocol-buffer definitions (e.g. `A.proto`, `B.proto` and `C.proto`) in a directory (e.g. `/data/my_proto_files`) where your top-level file (e.g. `A.proto`) imports at least one other definition
```
Comment (powersj) [inline plugins/parsers/xpath/README.md:34]: Does this path need to be absolute? If it is relative, does it need to be relative to the telegraf binary?

If the files are all in the same directory with the binary do you need to specify the full path or just '.'?

**Review by powersj — COMMENTED** [review]
Two questions about the paths, and then this is good to go!

Thank you or jumping on this!

**Review by powersj — APPROVED** [review]

**Comment by weknin** [comment]
Hey! This bug occurs on our end as well, any ETA for a merge ?

**Review by MyaLongmire — APPROVED** [review]

**Comment by weknin** [comment]
Thanks! :)

**Pull request merged**